### PR TITLE
add support for Integer dot product

### PR DESCRIPTION
**Pull request by rjodinchr**

### CODE REVIEW

**Comment by rjodinchr** [comment]
FYI @kpet

**Comment by lpavank** [comment]
@rjodinchr I might be missing a detail here, any particular reason (performance or functionality) for choosing arm_integer_dot_product vs khr_integer_dot_product extension?

**Comment by kpet** [comment]
I was going to ask the same question as Pavan :).

**Comment by rjodinchr** [comment]
The application is using `arm_dot`, not `dot`

**Comment by rjodinchr** [comment]
I've had support for khr_integer_dot_product.

File: lib/SPIRVProducerPass.cpp [hunk 1]
```diff
@@ -4514,6 +4677,24 @@ SPIRVID SPIRVProducerPassImpl::GenerateInstructionFromCall(CallInst *Call) {
...
+  case Builtins::kIDotAccSatPackedUUU:
+  case Builtins::kIDotAccSatPackedSSS:
+  case Builtins::kIDotAccSatPackedUSS:
+  case Builtins::kIDotAccSatPackedSUS:
+  case Builtins::kIDotPackedUUU:
+  case Builtins::kIDotPackedSSS:
+  case Builtins::kIDotPackedUSS:
+  case Builtins::kIDotPackedSUS:
+  case Builtins::kArmDotAcc: {
+    if (SpvVersion() >= SPIRVVersion::SPIRV_1_6) {
```
Comment (alan-baker) [inline lib/SPIRVProducerPass.cpp:4691]: It is supported via extension before SPIR-V 1.6. With 1.6 you no longer need to mention the extension.

**Review by alan-baker — CHANGES_REQUESTED** [review]

File: lib/SPIRVProducerPass.cpp (hunk 1 above)
Comment (rjodinchr) [reply lib/SPIRVProducerPass.cpp]: Done

File: lib/SPIRVProducerPass.cpp [hunk 2]
```diff
@@ -3196,6 +3200,10 @@ void SPIRVProducerPassImpl::GenerateModuleInfo() {
     addSPIRVInst<kExtensions>(spv::OpExtension, "SPV_EXT_descriptor_indexing");
   }
 
+  if (hasIntegerDot()) {
```
[the remark this answers is not in the document]
Comment (rjodinchr) [reply lib/SPIRVProducerPass.cpp]: Done

**Review by alan-baker — APPROVED** [review]

**Pull request merged**